### ISSUE TITLE
Scroll bar showing even after closing category

### ISSUE THREAD

**Issue by retrouser955** [nbudin/react-blockly#75]
Hi!

The scroll bar is still showing even after closing the category that introduced it!

![image](https://user-images.githubusercontent.com/70205403/175550071-aeed87ad-38dc-421a-becd-7f21ccb9a334.png)

my Javascrpt code is
```jsx
function App() {
  const [xml, setXml] = useState('<xml xmlns="https://developers.google.com/blockly/xml"></xml>')
  const [javascriptCode, setJavascriptCode] = useState(null)
  return (
    <BlocklyWorkspace 
    toolboxConfiguration={toolbox}
    initialXml={xml}
    onXmlChange={(xml) => {
      setXml(xml)
      console.log(xml)
    }}
    className="w-screen h-screen"
    onImportXmlError={(e) => { console.log(e) }}
    workspaceConfiguration={{
      grid: {
        spacing: 20,
        length: 3,
        colour: "#ccc",
        snap: true
      },
      theme: DarkTheme,
      renderer: 'zelos',
      zoom: {
        controls: true,
        startScale: 0.9,
        maxScale: 3,
        minScale: 0.3,
        scaleSpeed: 1.2
      },
      trashcan: true,
      maxTrashcanContents: 0,
      move: {
        scrollbars: true,
        drag: true,
        wheel: true
      },
      readOnly: false
    }}
    onWorkspaceChange={(workspace) => {
      setJavascriptCode(Blockly.JavaScript.workspaceToCode(workspace))
      console.log(javascriptCode)
    }}
    />
  );
}
```

**Comment by retrouser955** [comment]
Blockly version: 8.0.3

react-blockly version: 7.0.1

**Comment by retrouser955** [comment]
I sort of fixed it by using CSS and setting the scroll bar's display to none. Even though this is not the way to fix it
```js
.blocklyScrollbarBackground {
    display: none;
}
.blocklyScrollbarHandle {
    display: none;
}
```

**Comment by retrouser955** [comment]
@nbudin I am sorry about the delay as well and yes I have tried it with Blockly with React (not react-blockly) and the issue was still there. Blockly works perfectly on VUE but not React.

**Comment by nbudin** [comment]
@retrouser955 I've tried to reproduce this just now using create-react-app locally and I can't see the behavior you're talking about.  If you wouldn't mind posting a repo that reproduces this error, that'd be super helpful for me.  Thank you!

**Comment by Malix-Off** [comment]
- [Comment Reply](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1173128415) from @nbudin:
> [[…]](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1173128415) a repo that reproduces this error [[…]](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1173128415)

Hello!
I am @retrouser955's manager for the project we are facing this issue.
I advise that you check [this branch](https://github.com/scratch-for-discord/Web-Application_Frontend/tree/v3-react) of [this repo](https://github.com/scratch-for-discord/Web-Application_Frontend), we have an hosted version but we can't post it for now

**Comment by nbudin** [comment]
Thanks @Malix-Off!

I'm starting to suspect this is browser-specific.  I downloaded that repository and ran that branch locally using Chrome for Mac, and even when setting the system preferences to always show scroll bars and significantly lowering the browser window size, I can't get a scroll bar to appear on the toolbox at all.  Here's what I see:

![Screen Shot 2022-07-03 at 10 00 22 AM](https://user-images.githubusercontent.com/7477/177049823-9e22bd11-4827-4009-bc99-4694ab811961.png)

Could you please let me know what browser and OS you're using?

**Comment by retrouser955** [comment]
@nbudin as mentioned in [this message](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1169981794), I have disabled the scroll bar with CSS and to get it to appear, you will have to remove these lines of code from App.css
```js
.blocklyScrollbarBackground {
    display: none;
}
.blocklyScrollbarHandle {
    display: none;
}
```

**Comment by nbudin** [comment]
Oh, sorry about that - I see the issue now.

This looks to me like a bad interaction between Tailwind's root style (they have a global style on the `<svg>` element that sets it to display) and the way Blockly is rendering the flyout scrollbar.  Blockly is setting the HTML attribute `display="none"` on the `<svg>` element, but it looks like at least in Chrome, the global styling overrides that.

Blockly is already setting a `style` attribute on the `<svg>`, so it might be possible for it to set `display: none` in there instead of `display="none"`, which I think would work.

In the meantime, I found that this worked around the issue for me:

```css
.blocklyFlyoutScrollbar[display="none"] {
    display: none;
}
```

It's not pretty, but it should work.  I'm also going to notify the Blockly developers and point them at this thread, so that if they want to try implementing the CSS attribute-based solution I suggested above, they can.

**Comment by nbudin** [comment]
@Malix-Off @retrouser955 It turns out this is a known issue in Blockly, which is marked as `wontfix` over there.  I added a comment with some additional details.  There's a piece of CSS in the description of that issue that works around the bad interaction, which I think is probably a better solution than what I wrote above, since it should work across the board.

**Comment by Malix-Off** [comment]
- [Comment Reply](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359) from @nbudin:
> [[…]](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359) this is a known issue in Blockly, which is marked as wontfix over there [[…]](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359)

Could you link the issue please?

- [Comment Reply](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359) from @nbudin:
> [[…]](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359) I added a comment [[…]](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359)

Could you also link this comment please?

- [Comment Reply](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359) from @nbudin:
> [[…]](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359) There's a piece of CSS in the description of that issue that works around the bad interaction [[…]](https://github.com/nbudin/react-blockly/issues/75#issuecomment-1174168359)

Could you also link this comment please?

**Comment by nbudin** [comment]
@Malix-Off These are all found in the issue I linked directly above your comment: https://github.com/google/blockly/issues/5840

The workaround is in the top-level description of that issue.

**Comment by retrouser955** [comment]
Ah ok thank you and sorry for the late reply, I was offline for about 4 days.

**Comment by jh0l** [comment]
I was wondering why when the blockly flyout collapsed the scrollbar for the flyout was still visible. I'm glad I found this issue only after spending a couple hours trying to figure out the source of the problem 😅 I wish I was better at google.

**Comment by HologramSteve** [comment]
> .blocklyScrollbarBackground {
>     display: none;
> }
> .blocklyScrollbarHandle {
>     display: none;
> }

yo this worked for me, thanks!
